agent/config/builder.go

```diff
@@ -342,8 +342,8 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		}
 	}
 
-	advertiseAddr := bindAddr
-	if detect != nil {
+	advertiseAddr := b.makeIPAddr(b.expandFirstIP("advertise_addr", c.AdvertiseAddrLAN), bindAddr)
+	if ipaddr.IsAny(advertiseAddr) {
```

Comment:
I think you can end up in here with advertise set to `0.0.0.0` and bind set to non-ANY with `detect` set to `nil`, so we probably still need a `nil` check.

Reply:
fixed